src/lib/viewers/archive/ArchiveExplorer.js

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Internationalize from 'box-ui-elements/es/elements/common/Internationalize';
+import {
+    dateCellRenderer,
+    sizeCellRenderer,
+    fileNameCellRenderer,
+} from 'box-ui-elements/es/features/virtualized-table-renderers';
+import VirtualizedTable from 'box-ui-elements/es/features/virtualized-table';
+import { Column } from 'react-virtualized/dist/es/Table/index';
+
+class ArchiveExplorer extends React.Component {
+    static propTypes = {
+        data: PropTypes.arrayOf(
+            PropTypes.shape({
+                type: PropTypes.string,
+                absolute_path: PropTypes.string,
+                name: PropTypes.string,
+                modified_at: PropTypes.string,
+                size: PropTypes.number,
+                path_collection: PropTypes.shape({
+                    total_count: PropTypes.number,
+                    entries: PropTypes.arrayOf(
+                        PropTypes.shape({
+                            type: PropTypes.string,
+                            absolute_path: PropTypes.string,
+                            name: PropTypes.string,
+                        }),
+                    ),
+                }),
+                parent: PropTypes.string,
+                item_collection: PropTypes.shape({
+                    total_count: PropTypes.number,
+                    entries: PropTypes.arrayOf(
+                        PropTypes.shape({
+                            type: PropTypes.string,
+                            absolute_path: PropTypes.string,
+                            name: PropTypes.string,
+                        }),
+                    ),
+                }),
+            }),
+        ).isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+        const { cache, root } = this.buildCache(props.data);
+        this.cache = cache;
+        this.root = root;
+
+        this.state = {
+            currentData: this.getCurrentData(cache, root),
```

Comment:
This naming is kind of duplicative, since `this.state` is always the "current" data for the component. What does this data represent? A collection of items?